src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java

```diff
@@ -0,0 +1,310 @@
+                "/ResourceDirectory/parent/atom:link/@href", doc);
+        XMLAssert.assertXpathEvaluatesTo(FORMAT.format(myRes.lastmodified()),
+                "/ResourceDirectory/lastModified", doc);
+        XMLAssert.assertXpathEvaluatesTo("myres", "/ResourceDirectory/children/child/name", doc);
+        XMLAssert.assertXpathEvaluatesTo("http://localhost:8080/geoserver/rest/resources/mydir/myres", 
+                "/ResourceDirectory/children/child/atom:link/@href", doc);
+    }
+    
+    @Test
+    public void testDirectoryJSON() throws Exception {
+        JSON json = getAsJSON("/rest/resource/mydir?format=json");
+        //print(json);
+        String expected = "{\"ResourceDirectory\": {"
+                + "\"name\": \"mydir\","
+                + "\"parent\":   {"
+                + "  \"path\": \"/\","
+                + "    \"link\":     {"
+                + "      \"href\": \"http://localhost:8080/geoserver/rest/resources/?format=json\","
+                + "      \"rel\": \"alternate\","
+                + "      \"type\": \"application/json\""
+                + "  }"
+                + "},"
+                + "\"lastModified\": \"" + FORMAT.format(myRes.lastmodified()) + "\","
+                + "  \"children\": {\"child\": [  {"
+                + "    \"name\": \"myres\","
+                + "    \"link\":     {"
+                + "      \"href\": \"http://localhost:8080/geoserver/rest/resources/mydir/myres\","
+                + "      \"rel\": \"alternate\","
+                + "      \"type\": \"application/octet-stream\""
+                + "    }"
+                + "  }]}"
+                + "}}";
+        JSONAssert.assertEquals(expected, (JSONObject) json);
+    }
+    
+    @Test
+    public void testDirectoryHTML() throws Exception {
+        XMLUnit.setXpathNamespaceContext(NS_HTML);
+        Document doc = getAsDOM("/rest/resource/mydir?format=html");
+        //print(doc);
+        XMLAssert.assertXpathEvaluatesTo("Name: 'mydir'", "/x:html/x:body/x:ul/x:li[1]", doc);
+        XMLAssert.assertXpathEvaluatesTo("http://localhost:8080/geoserver/rest/resources/?format=html", 
+                "/x:html/x:body/x:ul/x:li[2]/x:a/@href", doc);
+        XMLAssert.assertXpathEvaluatesTo("Last modified: " + new Date(myRes.lastmodified()).toString(), 
+                "/x:html/x:body/x:ul/x:li[3]", doc);
+        XMLAssert.assertXpathEvaluatesTo("http://localhost:8080/geoserver/rest/resources/mydir/myres", 
+                "/x:html/x:body/x:ul/x:li[4]/x:ul/x:li/x:a/@href", doc);
+    }
+    
+    @Test
+    public void testDirectoryHeaders() throws Exception {
+        MockHttpServletResponse response = getAsServletResponse("/rest/resource/mydir?format=xml");
+        Assert.assertEquals(FORMAT_HEADER.format(myRes.lastmodified()), response.getHeader("Last-modified"));
+        Assert.assertEquals("/", response.getHeader("Resource-parent"));
```

Comment:
If your test fixture had more depth, you could test charset handling on the ```Resource-Parent``` header.